lines.go

```diff
@@ -46,9 +46,25 @@ func newConcepto(line *bill.Line, regime *tax.Regime) *Concepto {
 		ValorUnitario: line.Item.Price.String(),
 		Importe:       line.Sum.String(),
 		Descuento:     formatOptionalAmount(internal.TotalLineDiscount(line)),
-		ObjetoImp:     ObjetoImpSi,
+		ObjetoImp:     lineSubjectToTax(line, regime),
 		Impuestos:     newConceptoImpuestos(line, regime),
 	}
 
 	return concepto
 }
+
+func lineSubjectToTax(line *bill.Line, r *tax.Regime) string {
+	if len(line.Taxes) == 0 {
+		return ObjetoImpNo
+	}
+	// Check if the line's VAT is exempt from VAT
+	for _, combo := range line.Taxes {
+		if combo.Category == tax.CategoryVAT {
+			rate := r.Rate(combo.Category, combo.Rate)
+			if rate != nil && rate.Exempt {
```

Comment:
⚠️ Watch out here ⚠️: In Mexico, invoice lines can be either non-subject or subject of the tax, and those that are subject can be exempt or not. The value `Exento` should be set in the key `TipoValor` if that happens. That means we need to provide a way to distinguish between the non-subject and the exempt cases.

Reply:
right... so the best option here then would be to differentiate between lines that have taxes (subject) and those that don't (not-subject). I'll check the `TipoValor`, as there is also the case here of zero rated.

Reply:
Okay. So this is now fixed. The `TipoFactor` is now set appropriately. GOBL has also been updated to allow invoice lines without taxes, which implies the `ObjetoImp` will be set to `002`.

Reply:
@samlown in other regimes, we use the "exempt" rate for both and then an extension specifies whether is an exemption or a non-subject case:

* Portugal: https://github.com/invopop/gobl/blob/main/regimes/pt/extensions.go#L46
* Italy: https://github.com/invopop/gobl/blob/main/regimes/it/extensions.go#L100

Reply:
Previous comment updated to include TicketBAI

Reply:
yeah, they require explicit codes as they are more like VAT declarations and always require VAT to be present. But the case in Mexico is clear. The special code is explicit: "there are no taxes".

Reply:
Thanks, that makes sense! I just thought that a person generating GOBL in different countries, unaware of the particularities of each local target format, may find surprising having to treat, seemingly, the same circumstance so differently depending on the country.